src/test/java/core/basesyntax/service/impl/CsvDataReaderTest.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import org.junit.Test;
+
+public class CsvDataReaderTest {
+    private static final String INPUT_VALID_FILE =
+                                "src/test/test-resources/input-files/TestValidInputFile.csv";
+    private static final String INPUT_EMPTY_FILE =
+                                "src/test/test-resources/input-files/TestEmptyFile.csv";
+    private static final String INPUT_NONEXISTENT_FILE =
+                                "src/test/test-resources/input-files/doesNotExist.txt";
+    private static final String INPUT_RANDOM_LINES_FILE =
+                                "src/test/test-resources/input-files/TestRandomLines.csv";
+    private static final CsvDataReader csvDataReader = new CsvDataReader();
+
+    @Test
+    public void readFileLines_allValidConditions_Ok() {
+        List<String> actual = csvDataReader.readFileLines(INPUT_VALID_FILE);
+        List<String> expected = List.of("type,fruit,quantity", "b,banana,20");
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void readFileLines_EmptyFileOnInput_NotOk() {
+        assertThrows(RuntimeException.class, () -> csvDataReader.readFileLines(INPUT_EMPTY_FILE));
+    }
+
+    @Test
+    public void readFileLines_nonexistentFilePath_NotOk() {
+        assertThrows(RuntimeException.class,
+                    () -> csvDataReader.readFileLines(INPUT_NONEXISTENT_FILE));
+    }
+
+    @Test
+    public void readFileLines_inputWithCustomLines_Ok() {
+        List<String> actual = csvDataReader.readFileLines(INPUT_RANDOM_LINES_FILE);
+        List<String> expected = List.of("ugpwrevonrov",
+                                        "fwpueivwrnv",
+                                        "vpaerhgp48g  oiwjeiof",
+                                        "whvwvoiwn");
+        assertEquals(expected, actual);
+    }

```

Comment:
https://junit.org/junit5/docs/current/user-guide/#writing-tests-classes-and-methods
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/46851188/7c2b1fdc-6ee6-4e6e-bb82-0f8c04033d04)

Reply:
JUnit4 echo